l10n_eu_oss/hooks.py

```diff
@@ -0,0 +1,57 @@
+# -*- encoding: utf-8 -*-
+# Copyright 2021 Rodrigo Bonilla <rodrigo.bonilla@factorlibre.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, SUPERUSER_ID, _
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        _logger.info("UPDATE ACCOUNT TAX OSS")
+        account_tax_ids = env['account.tax'].search([
+            ('oss_country_id', '!=', False)])
+        for account_tax_id in account_tax_ids:
+            if not account_tax_id.base_code_id and not account_tax_id.ref_base_code_id:
+                code_bi = _upgrade_tax_code(env, account_tax_id, 'TB')
+                account_tax_id.write({
+                    "base_code_id": code_bi.id,
+                    "base_sign": 1,
+                    "ref_base_code_id": code_bi.id,
+                    "ref_base_sign": -1,
+                })
+            if not account_tax_id.tax_code_id and not account_tax_id.ref_tax_code_id:
+                code_c = _upgrade_tax_code(env, account_tax_id, 'C')
+                account_tax_id.write({
+                    "tax_code_id": code_c.id,
+                    "tax_sign": 1,
+                    "ref_tax_code_id": code_c.id,
+                    "ref_tax_sign": -1,
+                })
+
+
+def _upgrade_tax_code(env, account_tax_id, type):
+    account_tax_code = env["account.tax.code"]
```

Comment:
Reuse existing code instead of duplicating it.

Reply:
Done